File: C++/is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0;
        int j = 0;
        
        while (s[i] && t[j]) {
            if (s[i] == t[j])
            {
                i++;
            }
            j++;
        }
        
        return s[i] == '\0';        
    }
};

File: C#/is-subsequence.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        int i = 0;
        for (int j = 0; i < s.Length && j < t.Length; j++)
        {
            if (s[i] == t[j]) {
                i++;
            }
        }
        return i == s.Length;
    }
}